From e1a8156cec9c4f01636dc1088bc2266b08776a90 Mon Sep 17 00:00:00 2001
From: Erez_Zadok <ezk@cs.sunysb.edu>
Date: Tue, 7 Aug 2007 18:52:05 -0400
Subject: [PATCH] Unionfs: support incremental versions

Also: release Unionfs 2.1 (since code has gotten much more stable since 2.0)

Unionfs will now report its precise version number and the kernel it was
meant for, upon module loading.  This is useful for maintenance purposes:
users can report the precise release number of Unionfs, and it also helps
prevent running older versions when newer ones are intended, or mixing
versions among supported kernels.

Signed-off-by: Erez Zadok <ezk@cs.sunysb.edu>
---
 fs/unionfs/Makefile      |    4 ++++
 include/linux/union_fs.h |    1 -
 2 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/fs/unionfs/Makefile b/fs/unionfs/Makefile
index 7c18a3c..a97e236 100644
--- a/fs/unionfs/Makefile
+++ b/fs/unionfs/Makefile
@@ -1,3 +1,7 @@
+UNIONFS_VERSION="2.1 (for 2.6.22.1)"
+
+EXTRA_CFLAGS += -DUNIONFS_VERSION=\"$(UNIONFS_VERSION)\"
+
 obj-$(CONFIG_UNION_FS) += unionfs.o
 
 unionfs-y := subr.o dentry.o file.o inode.o main.o super.o \
diff --git a/include/linux/union_fs.h b/include/linux/union_fs.h
index 9bc4e3b..d13eb48 100644
--- a/include/linux/union_fs.h
+++ b/include/linux/union_fs.h
@@ -12,7 +12,6 @@
 #ifndef _LINUX_UNION_FS_H
 #define _LINUX_UNION_FS_H
 
-#define UNIONFS_VERSION  "2.0"
 /*
  * DEFINITIONS FOR USER AND KERNEL CODE:
  */
-- 
1.5.4.3

